src/dmd/e2ir.d

```diff
@@ -2512,6 +2512,36 @@ elem *toElem(Expression e, IRState *irs)
                 // we can skip the compare if the structs are empty
                 e = el_long(TYbool, ie.op == TOKidentity);
             }
+            else if (t1.ty == Tcomplex80 && Target.realpad != 0)
+            {
+                /* creal has padding in the middle on some platforms,
+                 * so do identity comparison on real and imaginary parts
+                 * separately.
```

Comment:
Please add comment illustrating what the following generated code looks like as an expression.